ttnn/cpp/ttnn/operations/eltwise/unary/device/kernels/compute/tanhshrink_sfpu_kernel.cpp

```diff
@@ -0,0 +1,57 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstdint>
+#include "compute_kernel_api/common.h"
+#include "compute_kernel_api/eltwise_binary.h"
+#include "compute_kernel_api/eltwise_binary_sfpu.h"
+#include "compute_kernel_api/tile_move_copy.h"
+#include "compute_kernel_api/eltwise_unary/eltwise_unary.h"
+#include "compute_kernel_api/eltwise_unary/sfpu_split_includes.h"
+#include "compute_kernel_api.h"
+#include "compute_kernel_api/copy_dest_values.h"
+
+namespace NAMESPACE {
+void MAIN {
+    uint32_t per_core_block_cnt = get_compile_time_arg_val(0);
+    uint32_t per_core_block_dim = get_compile_time_arg_val(1);
+
+    constexpr auto cb_input = tt::CBIndex::c_0;
+    constexpr auto cb_output = tt::CBIndex::c_2;
+    init_sfpu(cb_input, cb_output);
+
+    for (uint32_t block_index = 0; block_index < per_core_block_cnt; block_index++) {
+        cb_reserve_back(cb_output, per_core_block_dim);
+        for (uint32_t tile_index = 0; tile_index < per_core_block_dim; ++tile_index) {
+            cb_wait_front(cb_input, 1);
+            tile_regs_acquire();
+
+            // Pop tile after tile, copy to DST and pack
+            copy_tile_to_dst_init_short(cb_input);
+            copy_tile(cb_input, 0, 0);
+
+            tanh_tile_init();
+            tanh_tile(0);
+
+            // output = cb_input - tanh(x)
+            copy_dest_values(1, 0);
+            copy_tile_to_dst_init_short(cb_input);
+            copy_tile(cb_input, 0, 0);
+            sub_binary_tile_init();
+            sub_binary_tile(0, 1);
+
+            tile_regs_commit();
+
+            tile_regs_wait();
+
+            pack_tile(0, cb_output);
```

Comment:
I think you could avoid the `copy_dest_values` if you did this:
```suggestion
            copy_tile_init(cb_input);
            copy_tile(cb_input, 0, 1);
            sub_binary_tile_init();
            sub_binary_tile(1, 0);

            tile_regs_commit();

            tile_regs_wait();

            pack_tile(1, cb_output);
```

Reply:
We can’t implement this approach in SFPU because the destination index acts as the base, and it uses the destination offset to calculate the next tile starting from dst[0]. However, for FPU, you can simply swap the CB order.

Reply:
@mouliraj-mcw can you elaborate? Here's my understanding:
```cpp
copy_tile(cb_input, 0, 1); // cb_input[0] -> dst[1]
sub_binary_tile(1, 0); // dst[1] - dst[0] -> dst[1]
pack_tile(1, cp_output/*, output_tile_index = 0 */); // dst[1] -> cb_output[0]
```
I'm not sure why you say you can't implement this approach in SFPU. Have you actually tried the suggestion?

Reply:
Alternatively, you could perform `tanh_tile` on `dst[1]` and remove the `copy_dest_values` as well